ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1127,17 +1168,30 @@ private class PostgresLedgerDao(
 
 }
 
-object PostgresLedgerDao {
+object JdbcLedgerDao {
   def apply(
       dbDispatcher: DbDispatcher,
       contractSerializer: ContractSerializer,
       transactionSerializer: TransactionSerializer,
       valueSerializer: ValueSerializer,
-      keyHasher: KeyHasher): LedgerDao =
-    new PostgresLedgerDao(
+      keyHasher: KeyHasher,
+      dbType: JdbcLedgerDao.DbType): LedgerDao =
+    new JdbcLedgerDao(
       dbDispatcher,
       contractSerializer,
       transactionSerializer,
       valueSerializer,
-      keyHasher)
+      keyHasher,
+      dbType)
+
+  sealed trait DbType {
```

Comment:
To clarify my previous comment, I would make queries part of the implementation of this trait.

Reply:
The downside of this is that many queries will be duplicated, and modifying a query in a DB-specific trait makes it easy to not modify the respective query in all the other traits.

Reply:
For reference, I initially started out on the inheritance approach which duplicated a lot of code: https://github.com/digital-asset/daml/commit/754fd7b60985325571850c65c28171d35242d1a4#diff-3c124d60ef536a91fb6039a3d6b350aa and got worried about the maintainability of multiple DAOs.

But we should be able to come up with something more scoped, i.e. only overriding "non-standard" SQL statements. Let me take a stab at this and see how this turns out.

Another alternative could be to try to avoid Postgres-special syntax. For the "on conflict .. do nothing" this can mean standardizing on the H2db approach of adding a where clause filter to achieve the same effect. I did not do so since I was worried about causing perf regressions on the Postgres side.

Reply:
@stefanobaghino-da , I moved the Statement overrides into the trait. Please take a look: https://github.com/digital-asset/daml/pull/2425/commits/87be58d48544dab10c8d9432ae925aa773580725

Reply:
Yeah, I prefer this approach much more, thanks.

Reply:
I've added a single comment on that commit, feel free to fulfill the requirement or disregard it, it's not that important.